include/stl2/view/subrange.hpp

```diff
@@ -12,260 +12,245 @@
 #ifndef STL2_VIEW_ITERATOR_RANGE_HPP
 #define STL2_VIEW_ITERATOR_RANGE_HPP
 
+#include <tuple>
+#include <stl2/utility.hpp>
 #include <stl2/detail/fwd.hpp>
+#include <stl2/detail/algorithm/tagspec.hpp>
 #include <stl2/detail/iterator/concepts.hpp>
-#include <stl2/detail/range/access.hpp>
-#include <stl2/detail/range/concepts.hpp>
 #include <stl2/detail/iterator/dangling.hpp>
+#include <stl2/detail/range/concepts.hpp>
+#include <stl2/detail/range/primitives.hpp>
 #include <stl2/view/view_interface.hpp>
-#include <stl2/detail/algorithm/tagspec.hpp>
-#include <stl2/utility.hpp>
-
-#include <tuple>
 
 STL2_OPEN_NAMESPACE {
-	// __uncvref here is needed because of a libstdc++ bug.
 	template <class T>
-	concept bool _PairLike =
-		meta::Integral<std::tuple_size<__uncvref<T>>> &&
-		std::tuple_size<__uncvref<T>>::value == 2 &&
-		meta::Trait<std::tuple_element<0, __uncvref<T>>> &&
-		meta::Trait<std::tuple_element<1, __uncvref<T>>>;
+	concept bool _PairLike = requires(T t) {
+		typename std::tuple_size<T>;
+		requires meta::Integral<std::tuple_size<T>>;
+		requires std::tuple_size<T>::value == 2;
+		typename std::tuple_element<0, T>;
+		typename std::tuple_element<1, T>;
```

Comment:
No; and the answer's the same for `tuple_size` above. They may be incomplete, but they're always types since there are no constraints on the base template. This is too slavishly following the "always form types first in a typename constraint" rule.